internal/guest/snp/report.go

```diff
@@ -0,0 +1,321 @@
+//go:build linux
+// +build linux
+
+// package snp contains minimal functionality required to fetch
+// attestation reports inside an enlightened guest.
+package snp
```

Comment:
I think SNP is a more specific technology, maybe clearer to name this `amdsev` or `sev`? Then it can be a general package for interacting with /dev/sev.